build-python.sh

```diff
@@ -94,75 +94,79 @@ while [ $# -gt 0 ]; do
             then shift;
         fi
         BOOST_DIR="${1#*=}"
-        BUILD_ARGS="${BUILD_ARGS} --boost-dir='${BOOST_DIR}'"
+        BUILD_ARGS="${BUILD_ARGS} --config-setting=cmake.define.Boost_DIR='${BOOST_DIR}'"
```

Comment:
```--config-settings``` or ```--config-setting``` in this file?

Reply:
`--config-settings` when passed to `pip`, `--config-setting` when passed to `build` 😅 

https://github.com/microsoft/LightGBM/pull/5759#discussion_r1224471802